expression/builtin_string.go

```diff
@@ -385,40 +385,46 @@ type repeatFunctionClass struct {
 }
 
 func (c *repeatFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinRepeatSig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeString)
```

Comment:
@zz-jason 
This type inferring is not complete.
For example,
``` sql
create table t(
	c_int int,
	c_double double,
    c_decimal decimal,
    c_binary binary(20),
    c_char char(20),
    c_varchar varchar(20),
    c_varbinary varbinary(20),
    c_blob blob,
    c_date date,
    c_datetime datetime,
    c_enum enum('a', 'b', 'c')
)
```
``` sql
select repeat(c_char, 2), repeat(c_varchar,2 ) , repeat(c_binary,2 ), repeat(c_varbinary,2 ), repeat(c_blob,2 ) from t;
```
In MySQL :
<img width="787" alt="2017-06-27 14 44 16" src="https://user-images.githubusercontent.com/9039012/27574004-27dc3e0a-5b47-11e7-8e92-c35df6d56d4d.png">